### DISCUSSION TITLE
Custom Serialization Feature Problem

### DISCUSSION THREAD

**Discussion by casab** [ClickHouse/ch-go#317] [Q&A]
When I was checking the code for decoding, I've realized that if the protocol version supports custom serialization feature ch-go can't decode the data at all. Can someone explain to me the logic behind the following code?

https://github.com/ClickHouse/ch-go/blob/main/proto/results.go#L101

**Comment by ernado** [comment]
This code does following

1. If custom serialization is supported, try decode custom serialization flag
2. If flag is true, return error (i.e. custom serialization is used and we can't proceed)
3. If flag is false, continue decoding

So if protocol version supports custom serialization, ch-go can decode data.
But if custom serialization is used, the error is returned.

**Reply by casab** [reply]
Hmm interesting, we are not using any custom serialization. But for exact tables it's working fine for 23.6.2.18 while giving custom serialization error for 23.7.1.2470. Is it possible that ch-go is reading the wrong flag for custom serialization on newer version?

**Reply by ernado** [reply]
I've added `v23.7.2.25-stable` to e2e tests and everything looks ok.

But I've also experienced this error in my internal e2e tests, probably there is something in `23.7.1`?

**Reply by ernado** [reply]
There https://github.com/ClickHouse/ch-go/pull/318 is ok, so we need some way to reproduce this error now :thinking:

**Reply by casab** [reply]
Sadly I couldn't figure out a way to reproduce it. I was getting this error, but there was nothing special about the column at 10th index, it's just a Uint64 column, otherwise I was going to open an issue instead;
```panic: decode block: decode block: raw block: target: column [10] has custom serialization (not supported)```
